Added prebuild-check make target

kwk opened the pull request:
This change introduces a check target called `prebuild-check` which checks for required tools (`git`, `hg`, `go`, and `glide`) and also if the project is living inside the correct path which is `$GOPATH/src/$PACKAGE_NAME`.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/almighty/almighty-core/57)
<!-- Reviewable:end -->

aslakknutsen commented:
*[Makefile, line 61 \[r1\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNkzNAMYfm74yzXmpeI:-KNkzNAMYfm74yzXmpeJ:-164878593) ([raw file](https://github.com/almighty/almighty-core/blob/0d18837d3582217e1994d9f4dcedbaa0fc0dd21d/Makefile#L61)):*
> ```Makefile
> 
> .PHONY: clean-generated
> clean-generated: prebuild-check
> ```

Do I really need Go to clean?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

aslakknutsen commented:
*[Makefile, line 122 \[r1\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNkzYG8TDNKQaIwxvAy:-KNkzYG8TDNKQaIwxvAz:-1116483994) ([raw file](https://github.com/almighty/almighty-core/blob/0d18837d3582217e1994d9f4dcedbaa0fc0dd21d/Makefile#L122)):*
> ```Makefile
> endif
> # Check that the code is located in $GOPATH/src/github.com/almighty/almighty-core
> ifneq ($(GOPATH)/src/$(PACKAGE_NAME),$(shell pwd))
> ```

This assumes GOPATH is only a single location. It can be a List

``make prebuild-check 
Makefile:123: *** The code and this Makefile must live here "/home/aslak/.gvm/pkgsets/go1.6.2/almighty:/home/aslak/.gvm/pkgsets/go1.6.2/global/src/github.com/almighty/almighty-core" and not here "/home/aslak/.gvm/pkgsets/go1.6.2/almighty/src/github.com/almighty/almighty-core".  Stop.
`

```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

kwk commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[Makefile, line 61 \[r1\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNkzNAMYfm74yzXmpeI:-KNkzZmj0VgT-ccCGGXx:184475563) ([raw file](https://github.com/almighty/almighty-core/blob/0d18837d3582217e1994d9f4dcedbaa0fc0dd21d/Makefile#L61)):*
<details><summary><i>Previously, aslakknutsen (Aslak Knutsen) wrote…</i></summary>
> Do I really need Go to clean?
</details>
The check doesn't do any harm and should be modified so that it can run efficiently without disturbing anything. And yes, you need go to run clean. Because otherwise you would have nothing to clean. 

To keep things simple I'd have `prebuild-check` be executed all the time without any exception.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

kwk commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[Makefile, line 122 \[r1\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNkzYG8TDNKQaIwxvAy:-KNkzud7PUhjoO6LqSYn:-784473432) ([raw file](https://github.com/almighty/almighty-core/blob/0d18837d3582217e1994d9f4dcedbaa0fc0dd21d/Makefile#L122)):*
<details><summary><i>Previously, aslakknutsen (Aslak Knutsen) wrote…</i></summary>
> This assumes GOPATH is only a single location. It can be a List
> 
> ``make prebuild-check 
> Makefile:123: *** The code and this Makefile must live here "/home/aslak/.gvm/pkgsets/go1.6.2/almighty:/home/aslak/.gvm/pkgsets/go1.6.2/global/src/github.com/almighty/almighty-core" and not here "/home/aslak/.gvm/pkgsets/go1.6.2/almighty/src/github.com/almighty/almighty-core".  Stop.
> `
> 
> ```
</details>
You're right. I'll fix that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

tsmaeder commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNlCWyvayh-wMwsRLJn:-KNlCWyvayh-wMwsRLJo:-734790241) (no related file):*
Windows complains: 

Makefile:123: *** The code and this Makefile must live here "C:\Users\thomas\code\ALM/src/github.com/almighty/almighty-core" and not here "/cygdrive/c/Users/thomas/code/ALM/src/github.com/almighty/almighty-core".  Stop.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

kwk commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNlCWyvayh-wMwsRLJn:-KNlE5G5zmKJntChT6rP:1314270120) (no related file):*
<details><summary><i>Previously, tsmaeder (Thomas Mäder) wrote…</i></summary>
> Windows complains: 
> 
> Makefile:123: *** The code and this Makefile must live here "C:\Users\thomas\code\ALM/src/github.com/almighty/almighty-core" and not here "/cygdrive/c/Users/thomas/code/ALM/src/github.com/almighty/almighty-core".  Stop.
> 
</details>
I'm sick of windows.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

kwk commented:
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNlCWyvayh-wMwsRLJn:-KNlwmDAYu0gVbRQAlh_:1658099629) (no related file):*
<details><summary><i>Previously, kwk (Konrad Kleine) wrote…</i></summary>
> I'm sick of windows.
</details>
@tsmaeder: Fixed in 4ca8a5b3811c44cc59052ee070d12cef3d478690

---

*[Makefile, line 122 \[r1\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNkzYG8TDNKQaIwxvAy:-KNlwPD2tOxuokwLBsKi:-220309000) ([raw file](https://github.com/almighty/almighty-core/blob/0d18837d3582217e1994d9f4dcedbaa0fc0dd21d/Makefile#L122)):*
<details><summary><i>Previously, kwk (Konrad Kleine) wrote…</i></summary>
> You're right. I'll fix that.
</details>
Fixed in 4ca8a5b3811c44cc59052ee070d12cef3d478690 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

tsmaeder commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[Makefile, line 125 \[r3\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNlyTu_SRBa-SQGMbva:-KNlyTu_SRBa-SQGMbvb:2134523604) ([raw file](https://github.com/almighty/almighty-core/blob/ef4e261c2c53189682d1fe35c7ccc16e4597793d/Makefile#L125)):*
> ```Makefile
> endif
> 	# Check that the code is located in $GOPATH/src/github.com/almighty/almighty-core
> 	go build .make/check-gopath.go
> ```

If we write tools in go, wouldn't it make sense to put them in an installable package?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->

kwk commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[Makefile, line 125 \[r3\]](https://reviewable.io:443/reviews/almighty/almighty-core/57#-KNlyTu_SRBa-SQGMbva:-KNm0Z_MEhZSMbxQ1kBu:328151713) ([raw file](https://github.com/almighty/almighty-core/blob/ef4e261c2c53189682d1fe35c7ccc16e4597793d/Makefile#L125)):*
<details><summary><i>Previously, tsmaeder (Thomas Mäder) wrote…</i></summary>
> If we write tools in go, wouldn't it make sense to put them in an installable package?
</details>
In principle yes, but the check_gopath tool is just for building the install-able product.  So I'm not sure. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/almighty/almighty-core/57)*
<!-- Sent from Reviewable.io -->
